crypto/src/obfuscate.rs

```diff
@@ -0,0 +1,99 @@
+use crate::prelude::{ClientId, ConversationId};
+use hex;
+use log::kv::{ToValue, Value};
+use openmls::framing::Sender;
+use openmls::group::QueuedProposal;
+use openmls::prelude::Proposal;
+use sha2::{Digest, Sha256};
+use std::fmt::{Debug, Formatter};
+
+pub(crate) trait Obfuscate {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result;
+}
+
+impl Obfuscate for &ConversationId {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(hex::encode(compute_hash(self)).as_str())
+    }
+}
+
+impl Obfuscate for &ClientId {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(hex::encode(compute_hash(self)).as_str())
+    }
+}
+
+impl Obfuscate for &Proposal {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(match self {
+            Proposal::Add(_) => "Add",
+            Proposal::Update(_) => "Update",
+            Proposal::Remove(_) => "Remove",
+            Proposal::PreSharedKey(_) => "PreSharedKey",
+            Proposal::ReInit(_) => "ReInit",
+            Proposal::ExternalInit(_) => "ExternalInit",
+            Proposal::AppAck(_) => "AppAck",
+            Proposal::GroupContextExtensions(_) => "GroupContextExtensions",
+        })
+    }
+}
+
+impl Obfuscate for &QueuedProposal {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        f.write_str(match self.proposal {
+            Proposal::Add(_) => "Add",
+            Proposal::Update(_) => "Update",
+            Proposal::Remove(_) => "Remove",
+            Proposal::PreSharedKey(_) => "PreSharedKey",
+            Proposal::ReInit(_) => "ReInit",
+            Proposal::ExternalInit(_) => "ExternalInit",
+            Proposal::AppAck(_) => "AppAck",
+            Proposal::GroupContextExtensions(_) => "GroupContextExtensions",
+        })
+    }
+}
+
+impl Obfuscate for &Sender {
+    fn obfuscate(&self, f: &mut Formatter<'_>) -> core::fmt::Result {
+        match self {
+            Sender::Member(leaf_node_index) => f.write_fmt(format_args!("Member{}", leaf_node_index)),
+            Sender::External(external_sender_index) => f.write_fmt(format_args!("External{:?}", external_sender_index)),
+            Sender::NewMemberProposal => f.write_str("NewMemberProposal"),
+            Sender::NewMemberCommit => f.write_str("NewMemberCommit"),
+        }
+    }
+}
+
+pub(crate) struct Obfuscated<T> {
+    inner: T,
+}
```

Comment:
It's not wrong, but I'm curious why you used a named struct instead of a tuple struct here? This acts like a newtype, for which pattern I think tuple structs are the more common implementation.